lib/constructBodyDigestSignature.js

```diff
@@ -0,0 +1,60 @@
+const crypto = require('crypto')
+
+const constructBodyDigestSignature = ({
+  headers: {
+    digest,
+    'content-type': contentType,
+    'content-encoding': contentEncoding
+  },
+  body
+}) => {
+  const { formattedAlgorithm, digestAlgorithm, digestValue } = parseBodyDigestHeader(digest)
+
+  const calculatedDigest = calculateDigest({
+    algorithm: formattedAlgorithm,
+    message: parseContentType(contentType, contentEncoding)(body),
+    encoding: lookupEncoding(formattedAlgorithm)
+  })
+
+  if (calculatedDigest !== digestValue) {
+    throw new Error('Calculated Digest does not match Digest from Request Header')
+  }
+
+  return `${digestAlgorithm}=${calculatedDigest}`
+}
+
+const parseBodyDigestHeader = (requestDigestHeader) => {
+  // TODO:(jeff-sexton) - Parse for multiple digests in comma separated list and select one
```

Comment:
[nit] Todos should follow the below format with proper punctuation:

// TODO(jeff-sexton): Parse for multiple digests in comma separated list and select one.

This way they are easy to grep for.

Reply:
Thanks, I wasn't clear on the preferred format. Corrected.